app/models/host/base.rb

```diff
@@ -144,6 +144,9 @@ def import_facts(facts, source_proxy = nil)
         importer.import!
       end
 
+      facts_importer = Foreman::Plugin.fact_importer.get(type).new(self, facts)
```

Comment:
There is already an `importer` defined a couple of [lines before](https://github.com/rahulbajaj0509/foreman/blob/823e960a27b13b86ab9f596d1a8f383a0c607941/app/models/host/base.rb#L141)

Reply:
What Shim says, this must be imported just once. Also use telemetry wrappers.